src/main/java/billgates/entities/User.java

```diff
@@ -1,29 +1,42 @@
 package billgates.entities;
 
+import java.util.List;
 import java.util.Objects;
 
 // singleton
 public class User {
 
     private static User instance;
-
     private int id;
     private String name;
     private String password;
     private int billID;
     private int currentBillID;
+    private List<AbstractEntry> currentBill;
+
+    private User(int id, String name, String password, int billID) {
+        this.id = id;
+        this.name = name;
+        this.password = password;
+        this.billID = billID;
+    }
 
     public static User getInstance(int id, String name, String password, int billID) {
         if (instance == null)
             instance = new User(id, name, password, billID);
         return instance;
     }
 
-    private User(int id, String name, String password, int billID) {
-        this.id = id;
-        this.name = name;
-        this.password = password;
-        this.billID = billID;
+    /**
+     * Precondition: instance != null
+     * This function returns the current User instance or throw a runtime exception if the current user does not exist.
+     *
+     * @return The current User instance.
+     */
+    public static User getInstance() {
```

Comment:
I'm not sure about getting two methods which are both `getInstance`. 
In what scenario will we use the `getInstance` with parameters?

For example, if we know that a `User` already exists in runtime memory, then we can directly call `getInstance`, otherwise, we can first call the `User` constructor, then call the `getInstance` method? 

If what I'm saying doesn't make sense, then just mark this as resolved, because I'm not really sure how "Singleton design pattern" works XD